src/components/incidentsprovider.tsx

```diff
@@ -53,12 +63,52 @@ export const incidentsReducer = (state: IncidentsStateType, action: IncidentsAct
     return { ...state.lastModified, [pk]: new Date().getTime() };
   };
 
+  const findStoredIncidentIndex = (pk: Incident["pk"]): Index | null => {
+    if (pk in state._storedIndexByPk) return state._storedIndexByPk[pk];
+    return null;
+  };
+
   const findIncidentIndex = (pk: Incident["pk"]): Index | null => {
     if (pk in state._indexByPk) return state._indexByPk[pk];
     return null;
   };
 
   switch (action.type) {
+    // StoreAll is for saving incidents that ever appeared in view in between app refreshes,
+    // i.e. incidents that could be present in several table pages (not necessarily only
+    // in the one that is currently in user's view)
+    case IncidentsType.StoreAll: {
+      const newIncidents: Incident[] = action.payload;
+      const _storedIndexByPk = createIncidentsIndex(newIncidents);
+      const millis = new Date().getTime();
+      const lastModifiedInStore: { [pk: number]: number } = {};
+      newIncidents.forEach((incident: Incident) => {
+        lastModifiedInStore[incident.pk] = millis;
+      });
+      const allIncidents = [...state.storedIncidents, ...newIncidents].reduce((i: Incident[],
+                                                                               { pk,
+                                                                                 start_time,
+                                                                                 end_time,
+                                                                                 stateful,
+                                                                                 details_url,
+                                                                                 description,
+                                                                                 ticket_url,
+                                                                                 open,
+                                                                                 acked,
+                                                                                 level,
+                                                                                 source,
+                                                                                 source_incident_id,
+                                                                                 tags
+                                                                               }: Incident) => {
```

Comment:
there is some logic to lining up the arguments with the one at the top, but this many indentations seems excessive.
Would prefer if this was done in a nicer way.